docs/contributing/control-plane/development.md

```diff
@@ -0,0 +1,103 @@
+# Development Guide
+This document is intended to provide contributors with an introduction to developing the control plane of the Sedna project. 
+There are two components in our control plane: [GM(Global Manager) and LC(Local Controller)][framework].
+
+## Overview
+
+Sedna provides various utilities for development wrapped in `make`.
+
+Most scripts require more or less only `make` + `bash` on the host, and generally
+stick to POSIX utilities. Some scripts use `docker` e.g. for image building or
+to use docker images containing special tools.
+
+## Read Conventions Before Coding
+- [k8s coding convention]
+- [k8s api convention] if you want to add or update api.
+
+## Building
+Before continuing, You need to follow the [prerequisites installation guide] if you haven't done it yet.
+
+### Building Code
+
+```shell
+# build GM
+make WHAT=gm
+
+# build LC
+make WHAT=lc
+
+# build GM/LC both
+make
+```
+`_output/bin` will contain the freshly build binaries `sedna-gm` and `sedna-lc` upon a successful build.
+
+### Add or Update API
+If you add or update APIs in `pkg/apis/`, you need to run:
+1. run `make crds` to update the api definition
+1. run `kubectl replace -f build/crds/` to update your kubernetes environment
+1. run `bash hack/update-codegen.sh` to update client-go code in `pkg/client`
+
+### Add or Update Dependencies
+
+Run the following commands to update [Go Modules]:
+```
+go mod tidy
+go mod vendor
+
+# Or: hack/update-vendor.sh
+```
+Run `hack/update-vendor-licenses.sh` to update [vendor licenses](/LICENSES).
+
+### Running Sedna
+
+- [See here to run GM](debug-gm.md)
```

Comment:
Those links are not working properly

Reply:
Which links do you mean? debug-gm.md exists.